Assets/Global/Scripts/Player/States/StateBase.cs

```diff
@@ -93,13 +96,20 @@ public virtual void Attack(InputAction.CallbackContext ctx)
         }
     }
 
+    public virtual void Death()
```

Comment:
Is there a reason for giving all states this method, they never call,  rather than doing `SetState(states.Death)` in the OnDeath() of the player?